src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -130,9 +142,13 @@ public S3MetaRequest makeMetaRequest(S3MetaRequestOptions options) {
             requestFilePath = options.getRequestFilePath().toString().getBytes(UTF8);
         }
 
-        long credentialsProviderNativeHandle = 0;
-        if (options.getCredentialsProvider() != null) {
-            credentialsProviderNativeHandle = options.getCredentialsProvider().getNativeHandle();
+        AwsSigningConfig signingConfig = options.getSigningConfig();
+        boolean created_signing_config = false;
+        if(signingConfig == null) {
+            if (options.getCredentialsProvider() != null) {
```

Comment:
what happens if cred provider is null? no default signing config?
why is this flow different from make client?

Reply:
meta request options allow empty signing config as the client level signing config will be used. I added documentation to the options method.

Reply:
misunderstood your question, fixed.